src/tools/EpicsHostArch.pl

```diff
@@ -1,36 +1,109 @@
-# appropriate for the CPU that this version of Perl was built for.
-# Any arguments to the program will be appended with separator '-'
-# to allow flags like -gnu -debug and/or -static to be added.
-
-# Before Base has been built, use a command like this:
-#   bash$ export EPICS_HOST_ARCH=`perl src/tools/EpicsHostArch.pl`
-#
-# If Base is already built, use
-#   tcsh% setenv EPICS_HOST_ARCH `perl base/lib/perl/EpicsHostArch.pl`
-
-# If your architecture is not recognized by this script, please send
-# the output from running 'perl --version' to the EPICS tech-talk
-# mailing list to have it added.
-
 use strict;
+use warnings;
+
+use Getopt::Std;
+$Getopt::Std::STANDARD_HELP_VERSION = 1;
 
 use Config;
 use POSIX;
 
-print join('-', HostArch(), @ARGV), "\n";
+use FindBin qw($Bin);
+use lib ("$Bin/../../lib/perl", $Bin);
+
+use Pod::Usage;
+
+=head1 NAME
+
+EpicsHostArch.pl - Prints the current host architecture
+
+=head1 SYNOPSIS
+
+B<toEpicsArch.pl> [extension]
+
+B<toEpicsArch.pl> -g <GNU-like arch> [extension]
+
+B<toEpicsArch.pl> -h
+
+=head1 DESCRIPTION
+
+Returns an architecture name for EPICS_HOST_ARCH that should be
+appropriate for the CPU that this version of Perl was built for.
+Any arguments to the program will be appended with separator '-'
+to allow flags like -gnu -debug and/or -static to be added.
+
+Before Base has been built, use a command like this:
+
+C<bash$ export EPICS_HOST_ARCH=`perl src/tools/EpicsHostArch.pl`>
+
+If Base is already built, use:
+
+C<tcsh% setenv EPICS_HOST_ARCH `perl base/lib/perl/EpicsHostArch.pl`>
+
+If your architecture is not recognized by this script, please send
+the output from running C<perl --version> to the EPICS tech-talk
+mailing list to have it added.
+
+If the C<-g> option is provided with an argument, print the EPICS
+architecture corresponding to the given GNU architecture tuplet.
+
+=head1 OPTIONS
+
+B<toEpicsArch.pl> understands the following options:
 
-sub HostArch {
-    my $arch = $Config{archname};
+=over 4
+
+=item B<-h>
+
+Help, display this document as text.
+
+=item B<-g> <GNU arch tuplet>
+
+If specified, convert the given GNU architecture tuplet instead.
+
+=back
+
+=head1 EXAMPLES
+
+C<EpicsHostArch.pl>
+
+C<EpicsHostArch.pl static>
+
+C<EpicsHostArch.pl -g powerpc64-linux>
+
+C<EpicsHostArch.pl -g MSWin32-x64 static>
+
+=cut
+
+our ($opt_h);
+our ($opt_g);
+$opt_h = 0;
+
+sub HELP_MESSAGE {
+	pod2usage(-exitval => 2, -verbose => $opt_h * 3);
+}
+
+HELP_MESSAGE() if !getopts('hg:') || $opt_h;
+
+# Convert GNU-like architecture tuples (<cpu>-<system>)
+# to EPICS terminology (<system>-<cpu>)
+#
+# Documentation for GNU-like terminology:
+# - https://www.gnu.org/software/autoconf/manual/autoconf-2.65/html_node/System-Type.html#System-Type
+# - https://git.savannah.gnu.org/cgit/config.git/tree/
+#
+# Some examples from the Debian project:
+# - https://wiki.debian.org/Multiarch/Tuples
+sub toEpicsArch {
+	my $arch = shift;
```

Comment:
If the argument to this subroutine starts with `darwin` the routine will call `uname` on line 120 and attempt to use that host-specific data to determine the macOS CPU type. Thus running `EpicsHostArch.pl -g darwin-ppc` on a `linux-x86_64` box will probably return `darwin-x86_64` which is obviously wrong. On this macOS (`darwin-aarch64`) laptop, the Perl `$Config{archname}` configuration variable is `darwin-2level` which is why using `uname` is necessary.

Reply:
Good catch, I completely missed it